### PR TITLE
[image-picker][sharing] Small fixes while working on tutorial

### PR DESCRIPTION
**Pull request by brentvatne**
# Why

- Add permission methods to web image-picker.
- Add tests for `isAvailableAsync` on sharing.

# How

I had to add an expo-permissions dependency on expo-image-picker in order to call into the appropriate permissions methods, because we don't have the same module registry infrastructure on web that we use to call permissions on native in https://github.com/expo/expo/pull/5061.

# Test Plan

Run tests.

### CODE REVIEW

File: packages/expo-image-picker/package.json [hunk 1]
```diff
@@ -33,11 +33,14 @@
...
   "homepage": "https://docs.expo.io/versions/latest/sdk/imagepicker/",
   "jest": {
-    "preset": "expo-module-scripts/ios"
+    "preset": "expo-module-scripts"
   },
   "unimodulePeerDependencies": {
     "@unimodules/core": "*"
   },
+  "dependencies": {
+    "expo-permissions": "*"
```
Comment (EvanBacon) [inline packages/expo-image-picker/package.json]: isn't this a `unimodulePeerDependencies` thing?

File: packages/expo-sharing/src/ExpoSharing.web.ts [hunk 2]
```diff
@@ -13,6 +13,10 @@ export default {
     return 'ExpoSharing';
   },
   isAvailableAsync(): Promise<boolean> {
```
Comment (EvanBacon) [inline packages/expo-sharing/src/ExpoSharing.web.ts]:
```suggestion
  async isAvailableAsync(): Promise<boolean> {
```
then remove the `Promise.resolve(false)` in favor of `false`

File: packages/expo-sharing/src/ExpoSharing.web.ts [hunk 3]
```diff
@@ -13,6 +13,10 @@ export default {
...
+    if (typeof navigator === 'undefined') {
+      return Promise.resolve(false);
```
Comment (EvanBacon) [inline packages/expo-sharing/src/ExpoSharing.web.ts]:
```suggestion
      return false;
```

File: packages/expo-sharing/src/ExpoSharing.web.ts [hunk 4]
```diff
@@ -13,6 +13,10 @@ export default {
...
+    }
+
     return Promise.resolve(!!(navigator as Navigator).share);
```
Comment (EvanBacon) [inline packages/expo-sharing/src/ExpoSharing.web.ts]:
```suggestion
    return !!(navigator as Navigator).share;
```

File: packages/expo-image-picker/package.json (hunk 1 above)
Comment (brentvatne) [reply packages/expo-image-picker/package.json]: not actually used for anything

Comment (tsapeta) [reply packages/expo-image-picker/package.json]: @EvanBacon - just to clarify - if we're importing other unimodule from the JS code then the dependency needs to be under `dependencies` so it gets installed by npm/yarn.

Comment (tsapeta) [reply packages/expo-image-picker/package.json]: @brentvatne I think you should have add `unimodules-permissions-interface` here as well 🤔

**Pull request merged**